boards/raspberry_pi_pico_w/src/main.rs

```diff
@@ -0,0 +1,209 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OxidOS Automotive 2025.
+
+//! Tock kernel for the Raspberry Pi Pico W.
+
+#![no_std]
+#![no_main]
+#![deny(missing_docs)]
+
+use capsules_core::virtualizers::virtual_alarm::VirtualMuxAlarm;
+use kernel::component::Component;
+use kernel::debug;
+use kernel::hil::gpio::Configure;
+use kernel::platform::{KernelResources, SyscallDriverLookup};
+use kernel::{capabilities, create_capability};
+use kernel::{scheduler::round_robin::RoundRobinSched, syscall::SyscallDriver};
+use rp2040::chip::{Rp2040, Rp2040DefaultPeripherals};
+use rp2040::gpio::{RPGpio, RPGpioPin};
+use rp2040::pio_gspi::PioGSpi;
+use rp2040::timer::RPTimer;
+use rp2040::{dma, pio};
+
+mod io;
+
+kernel::stack_size! {0x1500}
+
+// State for loading and holding applications.
+// How should the kernel respond when a process faults.
+const FAULT_RESPONSE: capsules_system::process_policies::PanicFaultPolicy =
+    capsules_system::process_policies::PanicFaultPolicy {};
+
+type CYW4343xSpiBus = capsules_extra::cyw4343::spi_bus::CYW4343xSpiBus<
+    'static,
+    PioGSpi<'static>,
+    VirtualMuxAlarm<'static, rp2040::timer::RPTimer<'static>>,
+>;
+
+type CYW4343xDriver = capsules_extra::cyw4343::CYW4343x<
+    'static,
+    RPGpioPin<'static>,
+    VirtualMuxAlarm<'static, rp2040::timer::RPTimer<'static>>,
+    CYW4343xSpiBus,
+>;
```

Comment:
```suggestion
type CYW4343xHw = capsules_extra::cyw4343::CYW4343x<
    'static,
    RPGpioPin<'static>,
    VirtualMuxAlarm<'static, rp2040::timer::RPTimer<'static>>,
    CYW4343xSpiBus,
>;
```

Ending in `Driver` is for syscall drivers.